src/extensions.jl

```diff
@@ -99,7 +99,14 @@ function newstruct(T, xs...)
     flds = Any[convert(fieldtype(T, i), x) for (i,x) in enumerate(xs)]
     return ccall(:jl_new_structv, Any, (Any,Ptr{Cvoid},UInt32), T, flds, length(flds))
   else
-    newstruct!(initstruct(T), xs...)
+    x = initstruct(T)
```

Comment:
Can you add a link to [your comment](
https://github.com/MikeInnes/BSON.jl/issues/2#issuecomment-452204339) here?